addon/components/polaris-text-field.js

```diff
@@ -504,21 +515,88 @@ export default Component.extend({
     return inputType === 'number' && !disabled;
   }).readOnly(),
 
-  checkFocus() {
-    let { input, focused } = this.getProperties('input', 'focused');
+  normalizedValue: computed('value', function() {
+    let value = this.get('value');
+    return value !== null ? value : '';
+  }),
 
-    if (input && focused) {
-      // TODO this _seems_ to work in Polaris without the row below, but not for us (it does not apply focus class)
-      // look what's here....though this seems to do the job for now
-      this.set('focus', true);
-      input.focus();
+  characterCount: computed('normalizedValue.length', function() {
+    return this.get('normalizedValue.length') || 0;
+  }),
+
+  characterCountLabel: computed('maxLength', 'characterCount', function() {
+    let { maxLength, characterCount } = this.getProperties(
+      'maxLength',
+      'characterCount'
+    );
+
+    return maxLength
+      ? `${characterCount} characters of ${maxLength} used`
+      : `${characterCount} characters`;
+  }),
+
+  characterCountClassName: computed('multiline', function() {
+    let classNames = ['Polaris-TextField__CharacterCount'];
+
+    if (this.get('multiline')) {
+      classNames.push('Polaris-TextField__AlignFieldBottom');
     }
-  },
+
+    return classNames.join(' ');
+  }),
+
+  characterCountText: computed('maxLength', 'characterCount', function() {
+    let { maxLength, characterCount } = this.getProperties(
+      'maxLength',
+      'characterCount'
+    );
+
+    return !maxLength ? characterCount : `${characterCount}/${maxLength}`;
+  }),
+
+  focus: computed('focused', function() {
+    return this.get('focused') || false;
+  }),
 
   setInput() {
     this.set('input', document.querySelector(`[id='${this.get('id')}']`));
   },
 
+  handleButtonPress(onChange) {
+    let minInterval = 50;
+    let decrementBy = 10;
+    let interval = 200;
+
+    let onChangeInterval = () => {
+      if (interval > minInterval) {
+        interval -= decrementBy;
+      }
+
+      onChange();
+
+      this.set(
+        'buttonPressTimer',
+        window.setTimeout(onChangeInterval, interval)
```

Comment:
I wonder if we're better off using runloop-compatible timer functions here 🤔 This is probably fine in React-land but it's making me nervous in Ember-world 😅

Reply:
👍